src/apps/pulse/hooks/useGasTankBalance.ts

```diff
@@ -0,0 +1,93 @@
+import { useState, useEffect } from 'react';
+
+interface ChainBalance {
+  chainId: string;
+  balance: string;
+}
+
+interface GasTankBalanceResponse {
+  balance: {
+    [chainId: string]: ChainBalance;
+  };
+}
+
+interface UseGasTankBalanceReturn {
+  totalBalance: number;
+  chainBalances: ChainBalance[];
+  isLoading: boolean;
+  error: Error | null;
+  refetch: () => Promise<void>;
+}
+
+export function useGasTankBalance(
+  walletAddress: string | null
+): UseGasTankBalanceReturn {
+  const [totalBalance, setTotalBalance] = useState<number>(0);
+  const [chainBalances, setChainBalances] = useState<ChainBalance[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<Error | null>(null);
+
+  const paymasterUrl = import.meta.env.VITE_PAYMASTER_URL;
+
+  const fetchGasTankBalance = async () => {
+    if (!walletAddress || !paymasterUrl) {
+      setTotalBalance(0);
+      setChainBalances([]);
+      return;
+    }
+
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      const response = await fetch(
+        `${paymasterUrl}/getGasTankBalance?sender=${walletAddress}`,
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        }
+      );
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch gas tank balance: ${response.status}`);
```

Comment:
Is this shown to the user? If so please instruct the user what to do next

Reply:
added a generic error